### PR TITLE
Add Bolt as a Payment Method Option (Solidus 3.1)

### PR DESCRIPTION
**Pull request by piyushswain**
This PR adds bolt as a payment method option during solidus installation.

The `install_generator.rb` file has been updated to add Bolt as an optional payment method during solidus installation.
The Bolt option in the payment method installs the `solidus_bolt` gem.

Ref https://github.com/nebulab/solidus_bolt/issues/109

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [ ] I have added automated tests to cover my changes.
- [ ] I have attached screenshots to demo visual changes.
- [ ] I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- [ ] I have updated the readme to account for my changes.

### CODE REVIEW

File: core/lib/generators/solidus/install/install_generator.rb [hunk 1]
```diff
@@ -159,6 +148,21 @@ def install_payment_method
       end
     end
 
+    def install_auth_plugin
+      # Doesn't need to add solidus_auth_devise if solidus_bolt is being installed as it already has the gem
```
Comment (DanielePalombo) [inline core/lib/generators/solidus/install/install_generator.rb:152]: Solidus_auth_devise is also present on [solidus_paypal_commerce_platform](https://github.com/solidusio-contrib/solidus_paypal_commerce_platform/blob/master/Gemfile#L15)? Why we didn't have to skip this part for that gem?

Comment (DanielePalombo) [reply core/lib/generators/solidus/install/install_generator.rb:152]: Now I see that we have included solidus_auth_devise [here](https://github.com/solidusio-contrib/solidus_bolt/blob/master/solidus_bolt.gemspec#L37). Do we really need it?

Comment (DanielePalombo) [reply core/lib/generators/solidus/install/install_generator.rb:152]: I have checked the [commit](https://github.com/solidusio-contrib/solidus_bolt/commit/043fb7908d74c6340bdfcccab04faffa3667d26d), and it seems a typo.

Comment (piyushswain) [reply core/lib/generators/solidus/install/install_generator.rb:152]: I'm not sure about that but if this is left unchecked then the installer gives the error that the same gem cannot be installed twice for this application

Comment (piyushswain) [reply core/lib/generators/solidus/install/install_generator.rb:152]: We can maybe remove the `solidus_auth_devise` gem from the gemspec and check if everything is working or not.

If everything works we won't need this extra logic here in the installer as well

**Comment by waiting-for-dev** [comment]
Closing, as we're only going to add it in the latest version (it doesn't make sense to start a new project with an old version).

**Pull request closed without merging**